intrpvar.h

```diff
@@ -748,18 +748,12 @@ PERLVAR(I, padix_floor,	PADOFFSET)	/* how low may inner block reset padix */
 #if defined(USE_POSIX_2008_LOCALE) && defined(MULTIPLICITY)
 PERLVARI(I, cur_locale_obj, locale_t, NULL)
 #endif
-#ifdef USE_PL_CURLOCALES
 
 /* This is the most number of categories we've encountered so far on any
  * platform, doesn't include LC_ALL */
 PERLVARA(I, curlocales, 12, const char *)
-
-#endif
-#ifdef USE_PL_CUR_LC_ALL
-
 PERLVARI(I, cur_LC_ALL, const char *, NULL)
 
-#endif
```

Comment:
Won't this change the layout for a perl built with `-DNO_LOCALE`?

Even with that fixed I'm hesitant to backport this change - I'm not sure it qualifies under the conditions from perlpolicy, and it appears to be a minor enhancement at best.

Reply:
#21366, which this fixes, is a regression introduced in 5.38.  Why would that not be something we need to fix?  I don't understand about this being an "enhancement".  It is a bare minimum fix.

I believe you are right about `NO_LOCALE`, and it is fixable as you surmised.  I would want to make sure there is no other combination that would need to be addressed.

Reply:
Sorry, I should have followed through the reference you provided.

I think it's reasonable to apply this is the bincompat is fixed.

I do worry about systems that don't have a querylocale() work-a-like (are there any? musl provides this extension) -- if I understand the issue this won't fix such systems.